client/ayon_deadline/plugins/publish/maya/submit_maya_cache_deadline.py

```diff
@@ -0,0 +1,169 @@
+import os
+from datetime import datetime
+
+from dataclasses import dataclass, field, asdict
+import pyblish.api
+from ayon_core.lib import (
+    is_in_tests,
+)
+from ayon_core.pipeline import (
+    tempdir,
+    AYONPyblishPluginMixin
+)
+from ayon_core.pipeline.publish.lib import (
+    replace_with_published_scene_path
+)
+from ayon_deadline import abstract_submit_deadline
+
+
+@dataclass
+class MayaPluginInfo:
+    ScriptJob: bool = field(default=True)
+    SceneFile: bool = field(default=None)   # Input
+    ScriptFilename: str = field(default=None)
+    Version: str = field(default=None)  # Mandatory for Deadline
+    ProjectPath: str = field(default=None)
+
+
+class MayaCacheSubmitDeadline(abstract_submit_deadline.AbstractSubmitDeadline,   # noqa
+                              AYONPyblishPluginMixin):
+    """Submit Maya scene to perform a local publish in Deadline.
+
+    Publishing in Deadline can be helpful for scenes that publish very slow.
+    This way it can process in the background on another machine without the
+    Artist having to wait for the publish to finish on their local machine.
+    """
+
+    label = "Submit Scene to Deadline (Maya)"
+    order = pyblish.api.IntegratorOrder
+    hosts = ["maya"]
+    families = ["publish.farm"]
+    targets = ["local"]
+    settings_category = "deadline"
+
+    def get_job_info(self, job_info=None):
+        instance = self._instance
+        context = instance.context
+        assert all(
+            result["success"] for result in context.data["results"]
+        ), "Errors found, aborting integration.."
```

Comment:
@iLLiCiTiT always tells me to avoid `assert` and raise an actual error accordingly. :) So I guess I'll say that here too.

Reply:
That is true `PublishError` should be used instead. But on another note, how this can happen? Why we're checking for failed instances inside plugin?

Reply:
> That is true PublishError should be used instead. But on another note, how this can happen? Why we're checking for failed instances inside plugin? If there would be failed instance, it would not get here, or?

It would in headless mode ;) it's our publisher UI stopping it (is it?)... not `pyblish`. Pyblish stops after validation, but if an error happens during Extraction... I believe default behavior is that it doesn't directly stop but may continue. As such, if this were running through CLI it may still get triggered.

Reply:
I'd say so, yes.